### DISCUSSION TITLE
What is darkreader looking for when detecting if website has native dark mode?

### DISCUSSION THREAD

**Discussion by sarunluitel** [darkreader/darkreader#8982] [Q&A]
I am building a site and have implemented my own dark mode. I was happy to read the blog post - 

Detecting website's dark theme in version 4.9.47 
https://darkreader.org/blog/detect-dark-theme/

It's not working well on youtube as of writing this. What is dark reader looking for on a site to qualify it as native dark mode and leave it alone. Are there some  meta headers, property on manifeat to help dark reader detect native dark mode

**Comment by alexanderby** [comment]
Hi @sarunluitel! It takes the computed backgrounds of `<html>` and `<body>` elements and makes the decision. On the first run it checks if there is some background color set through style attribute for these elements. If not it waits for the first style sheet to appear.

So I think putting something like `<html style="background: black">` will be the fastest way to disable Dark Reader for a website (if such option is switched on, but we can consider making it enabled by default). Also if your website is always dark, we can put it into a special list.

The code is [here](https://github.com/darkreader/darkreader/blob/master/src/inject/detector.ts).
